eZ/Bundle/EzPublishCoreBundle/Matcher/ViewMatcherRegistry.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Bundle\EzPublishCoreBundle\Matcher;
+
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+use eZ\Publish\Core\MVC\Symfony\Matcher\MatcherInterface;
```

Comment:
This interface is deprecated since 6.0. Instead, `eZ\Publish\Core\MVC\Symfony\Matcher\ViewMatcherInterface` should be used. View matchers that implement only the `ViewMatcherInterface` cause a fatal error:

```
PHP Fatal error:  Uncaught Symfony\Component\Debug\Exception\FatalThrowableError: Argument 2 passed to eZ\Bundle\EzPublishCoreBundle\Matcher\ViewMatcherRegistry::setMatcher() must be an instance of eZ\Publish\Core\MVC\Symfony\Matcher\MatcherInterface, instance of Netgen\Bundle\SiteBundle\Core\MVC\Symfony\Matcher\ContentBased\Id\Content given
```

Reply:
> This interface is deprecated since 6.0. Instead, `eZ\Publish\Core\MVC\Symfony\Matcher\ViewMatcherInterface` should be used. View matchers that implement only the `ViewMatcherInterface` cause a fatal error:
> 
> ```
> PHP Fatal error:  Uncaught Symfony\Component\Debug\Exception\FatalThrowableError:
> Argument 2 passed to eZ\Bundle\EzPublishCoreBundle\Matcher\ViewMatcherRegistry::setMatcher()
> must be an instance of eZ\Publish\Core\MVC\Symfony\Matcher\MatcherInterface, instance of
> Netgen\Bundle\SiteBundle\Core\MVC\Symfony\Matcher\ContentBased\Id\Content given
> ```

Thanks for pointing this out. It's being rectified via #2755.